Alternate implementation w/multiple error codes.

serac opened the pull request.

dfish3r commented:
Not API compatible.
```
***! MODIFIED CLASS: PUBLIC org.passay.RuleResultDetail  (not serializable)
        ===  CLASS FILE FORMAT VERSION: 52.0 <- 52.0
        ---! REMOVED FIELD: PROTECTED(-) FINAL(-) java.lang.String errorCode
```

dfish3r commented on src/main/java/org/passay/RuleResultDetail.java:
```diff
@@ -30,19 +31,52 @@ public RuleResultDetail(final String code, final Map<String, Object> params)
+   * @param  params  error details.
+   * @param  codes  One or more error codes. Codes MUST be provided in order of decreasing specificity.
+   */
+  public RuleResultDetail(final Map<String, Object> params, final String... codes)
```
I'd probably forgo varargs here just to keep symmetry with the other map. Plus it just feels like the codes are the first order concern and should come first.

serac replied:
Fixed.

dfish3r commented on src/main/java/org/passay/RuleResultDetail.java:
```diff
@@ -71,6 +105,6 @@ public String getErrorCode()
   public String toString()
   {
-    return String.format("%s:%s", errorCode, parameters);
+    return String.format("%s:%s", Arrays.asList(errorCodes), parameters);
```
`Arrays#toString` is probably better suited.

serac replied:
Fixed.

serac commented:
Addressed code review feedback. I can't comment on your API incompatible feedback. It's obviously API compatible in this sense of it has the same public API, but maybe you mean it _ABI_ incompatible, which I can see. That doesn't seem like an issue, but you know more about this than I do so I defer to your judgement.

dfish3r commented:
I've seen this called 'API Binary Compatibility'. Whatever it's called, it means you don't have to recompile your code to use the next revision of library. Which is not the case here. This change bumps the version to 1.6.0, can you update the pom in this PR?

serac commented:
Bumped version
